library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java

```
@@ -52,8 +52,14 @@ int getBatchStatus(long batchId) {
             int statusColumnIndex = cursor.getColumnIndexOrThrow(COLUMN_STATUS);
 
             while (cursor.moveToNext()) {
-                int status = cursor.getInt(statusColumnIndex);
-                statusCounts.put(status, statusCounts.get(status) + 1);
+                int statusCode = cursor.getInt(statusColumnIndex);
+
+                if (Downloads.Impl.isStatusError(statusCode)) {
```

:+1: fail fast in case of an error